Hymns/Infrastructure/Data/PdfPreloader.swift

```diff
@@ -0,0 +1,38 @@
+import Combine
+import Foundation
+import PDFKit
+import Resolver
+
+/**
+ * Asynchronously loads and caches PDF documents loaded from web URLs
+ */
+class PDFLoader {
+
+    private let analytics: AnalyticsLogger
```

Comment:
oops, I left the analytics object in. You can remove that or add some analytics if you want to learn how to do that. Just followed the pattern in AnalyticsLogger